Backend/src/test/java/ca/mcgill/cooperator/service/CooperatorServiceCourseTests.java

```diff
@@ -41,6 +41,18 @@ public void testCreateCourse() {
         assertEquals(name, courseService.getAllCourses().get(0).getName());
     }
 
+    @Test
+    public void testCourseUniqueName() {
+        String name = "ECSE321";
+        try {
+            courseService.createCourse(name);
+            // name must be unique so expect a SQLException
+            courseService.createCourse(name);
+        } catch (Exception e) {
+            assertEquals(1, courseService.getAllCourses().size());
```

Comment:
maybe double check that the error message is equal to what we want it to be

Reply:
The error message is `could not execute statement; SQL [n/a]; constraint [uk_c0r9atamxvbhjjvy5j8da1kam]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement` which is hard to check for equality since the constraint changes every time, so I'll just leave it as is